tests/conftest.py

```diff
@@ -82,10 +82,26 @@ def inner(  # pylint: disable=too-many-arguments,missing-docstring,useless-suppr
             evaluate=evaluate
         )
 
+        if isinstance(result_node.outputs.optimal_process_input, orm.BaseType):
+            optimal_process_input = result_node.outputs.optimal_process_input.value
+        elif isinstance(result_node.outputs.optimal_process_input, orm.List):
+            optimal_process_input = result_node.outputs.optimal_process_input.get_list()
+        else:
+            raise ValueError(
+                f'`optimal_process_input` {result_node.outputs.optimal_process_input} is not of a supported AiiDA type for testing'
+            )
+
         assert 'optimal_process_uuid' in result_node.outputs
         assert np.isclose(result_node.outputs.optimal_process_output.value, f_exact, atol=ftol)
+        assert np.allclose(type(x_exact)(optimal_process_input), x_exact, atol=xtol)
+
         calc = orm.load_node(result_node.outputs.optimal_process_uuid.value)
         assert np.allclose(type(x_exact)(input_getter(calc.inputs)), x_exact, atol=xtol)
+        assert np.allclose(
```

Comment:
Here I wonder if we could also check `input_getter(calc.inputs).uuid == optimal_process_input.uuid` - it's supposed to be exactly the same node.

Reply:
Nested inputs are not AiiDA nodes when they go in, so `input_getter(calc.inputs)` doesn't seem to have a uuid in that case.

Reply:
Aah right - keep it as is then.